server/src/main/java/org/tctalent/server/service/db/impl/DataSharingServiceImpl.java

```diff
@@ -38,358 +37,344 @@
 
-            for (PartnerDatabaseDefinition destination : destinations) {
+      for (PartnerDatabaseDefinition destination : destinations) {
 
-                log.info("Copying to " + destination.getCountry());
+        log.info("Copying to " + destination.getCountry());
 
-                try {
-                    List<PartnerTableDefinition> defs = destination.getTables();
+        try {
+          List<PartnerTableDefinition> defs = destination.getTables();
 
-                    log.info("Create local copy");
-                    for (PartnerTableDefinition def : defs) {
-                        copyTable(def);
-                    }
+          log.info("Create local copy");
+          for (PartnerTableDefinition def : defs) {
+            copyTable(def);
+          }
 
-                    postProcess();
+          postProcess();
 
-                    log.info("Export/Import remote copy");
-                    try (Connection tbbRemoteCopy = destination.connect()) {
-                        for (PartnerTableDefinition def : defs) {
-                            exportImport(tbbRemoteCopy, def);
-                        }
-                    }
-                } catch (Exception ex) {
-                    ok = false;
-                    reportError("Failed to copy to destination: " + destination.getCountry(), ex);
-                }
+          log.info("Export/Import remote copy");
+          try (Connection tbbRemoteCopy = destination.connect()) {
+            for (PartnerTableDefinition def : defs) {
+              exportImport(tbbRemoteCopy, def);
             }
+          }
         } catch (Exception ex) {
-            reportError("Exception performing data sharing", ex);
+          ok = false;
+          reportError("Failed to copy to destination: " + destination.getCountry(), ex);
         }
-
-        disconnect();
-
-        return ok;
+      }
+    } catch (Exception ex) {
+      reportError("Exception performing data sharing", ex);
     }
 
-    private void connect() throws SQLException {
-        //This is the default h2 user and password
-        String dbLocalCopyUser = "sa";
-        String dbLocalCopyPassword = "";
+    disconnect();
 
-        tbbLocalCopy = DriverManager.getConnection(DB_LOCAL_COPY_URL, dbLocalCopyUser, dbLocalCopyPassword);
-        tbbMaster = DriverManager.getConnection(masterJdbcUrl, masterUser, masterPwd);
+    return ok;
+  }
 
-    }
+  private void connect() throws SQLException {
+    //This is the default h2 user and password
+    String dbLocalCopyUser = "sa";
+    String dbLocalCopyPassword = "";
 
-    private void disconnect() {
-        try {
-            if (tbbMaster != null) {
-                tbbMaster.close();
-                tbbMaster = null;
-            }
-        } catch (Exception ex) {
-            reportError("Could not close master DB connection", ex);
-        }
+    tbbLocalCopy = DriverManager.getConnection(DB_LOCAL_COPY_URL, dbLocalCopyUser, dbLocalCopyPassword);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected a hardcoded database connection password in your code. This could allow anyone with access to your server to perform privileged actions on the database. We recommend that you avoid hardcoded passwords when connecting to a database and instead use environment variables or a configuration file to set the password.

[Learn more](https://cwe.mitre.org/data/definitions/259.html)

Suggested remediation:
Do not hardcode database password in Java code
```
@@ -156,3 +156,3 @@
     tbbLocalCopy = DriverManager.getConnection(DB_LOCAL_COPY_URL, dbLocalCopyUser, dbLocalCopyPassword);
-    tbbMaster = DriverManager.getConnection(masterJdbcUrl, masterUser, masterPwd);
+    tbbMaster = DriverManager.getConnection(masterJdbcUrl, masterUser, System.getenv(&#34;PASSWORD_VALUE&#34;));
 

```